client.go

```diff
@@ -1046,11 +1047,7 @@ func (f *File) Seek(offset int64, whence int) (int64, error) {
 	case os.SEEK_CUR:
 		f.offset = uint64(int64(f.offset) + offset)
 	case os.SEEK_END:
-		fi, err := f.Stat()
-		if err != nil {
-			return int64(f.offset), err
-		}
-		f.offset = uint64(fi.Size() + offset)
+		f.offset = uint64(f.size + offset)
```

Comment:
Using this cached size, we run into a problem. Follow these steps, create a new file (`f.size == 0`), now write out 512 bytes to that file. Seek to the end of the file. You are now at the beginning of the file, `(f.size + 0)` not the end `(f.Stat().Size() + 0)`.